File: BlackJack.ViewModels/GameViews/GetAllStepOfBotsGameView.cs
using BlackJack.ViewModels.EnumViews;
using System;
using System.Collections.Generic;

namespace BlackJack.ViewModels.GameViews
{
    public class GetAllStepOfBotsGameView
    {
        public List<BotStepGetAllStepOfBotsViewItem> BotSteps { get; set; }

        public GetAllStepOfBotsGameView()
        {
            BotSteps = new List<BotStepGetAllStepOfBotsViewItem>();
        }
    }

    public class BotStepGetAllStepOfBotsViewItem
    {
        public Guid Id { get; set; }
        public SuiteTypeEnumView Suite { get; set; }
        public RankTypeEnumView Rank { get; set; }
        public BotGetAllStepOfBotsView Bot { get; set; }
    }

    public class BotGetAllStepOfBotsView
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public decimal Balance { get; set; }
        public decimal Bet { get; set; }
    }
}

File: BlackJack.WEB/ClientApp/src/app/shared/entities/game.views/get-all-step-of-bots.game.view.ts
import { SuiteType } from '../../enums/suite-type';
import { RankType } from '../../enums/rank-type';

export class GetAllStepOfBotsGameView {
  botSteps?: Array<BotStepGetAllStepOfBotsViewItem>;

  constructor() {
    this.botSteps = new Array<BotStepGetAllStepOfBotsViewItem>();
  }
}
export class BotStepGetAllStepOfBotsViewItem {
  id?: string;
  suite?: SuiteType;
  rank?: RankType;
  bot?: BotGetAllStepOfBotsView;

  constructor() {
    this.bot = new BotGetAllStepOfBotsView();
  }
}

export class BotGetAllStepOfBotsView {
  id?: string;
  name?: string;
  balance?: number;
  bet?: number;
}
